### PR TITLE
[INLONG-5808][SDK] Sort SDK supports parse InlongMsg

### LINKED ISSUE #5808: [Feature][SDK] SortSdk support InlongMsg
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5808

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]
LGTM

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java [hunk 1]
```diff
@@ -48,12 +55,20 @@ public class MessageDeserializer implements Deserializer {
...
+    private static final char INLONGMSG_ATTR_ENTRY_DELIMITER = '&';
+    private static final char INLONGMSG_ATTR_KV_DELIMITER = '=';
+
     public MessageDeserializer() {
     }
 
     @Override
     public List<InLongMessage> deserialize(ClientContext context, InLongTopic inLongTopic, Map<String, String> headers,
-            byte[] data) throws Exception {
+                                           byte[] data) throws Exception {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:71]: It is not necessary to indent with the above parameters.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java [hunk 2]
```diff
@@ -144,4 +162,64 @@ private List<InLongMessage> transformMessageObjs(ClientContext context, InLongTo
...
+        String groupId = Optional.ofNullable(headers.get(INLONGMSG_ATTR_GROUP_ID))
+                .orElseThrow(() -> new IllegalArgumentException("Could not find "
+                        + INLONGMSG_ATTR_GROUP_ID + " in attributes!"));
+
+        InLongMsg inLongMsg = InLongMsg.parseFrom(msgBytes);
+        for (String attr : inLongMsg.getAttrs()) {
+            Map<String, String> attributes = StringUtil.splitKv(attr, INLONGMSG_ATTR_ENTRY_DELIMITER,
+                    INLONGMSG_ATTR_KV_DELIMITER, null, null);
+
+            /*String groupId = Optional.ofNullable(attributes.get(INLONGMSG_ATTR_GROUP_ID))
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:182]: Suggests removing unused codes.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:71]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:182]: fixed, thx

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java [hunk 3]
```diff
@@ -165,6 +169,33 @@ private void testDeserializeVersion1CompressionType2() {
...
     }
 
+    private void testDeserializeVersion2NoCompress() {
+        try {
+            String groupId = "sort_sdk_test_group_id";
+            String streamId = "sort_sdk_test_stream_id";
+            String attr = "m=0";
+            String ip = "1.2.3.4";
+            long dt = System.currentTimeMillis();
+            StringBuilder newAttrBuffer = new StringBuilder(attr);
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java:179]: Why name it `buffer`?

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java [hunk 4]
```diff
@@ -97,6 +98,9 @@ public void testDeserialize() {
 
         //5. testDeserializeVersion1CompressionType2
         testDeserializeVersion1CompressionType2();
+
+        //6. DeserializeVersion2NoCompress
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java:102]: One blank after `//` please.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java [hunk 5]
```diff
@@ -106,4 +245,26 @@ public static String formatDate(Date date) {
...
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMddHH");
+                Date date = simpleDateFormat.parse(value.substring(0, 10));
+                return new Timestamp(date.getTime()).getTime();
+            } else {
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMddHHmm");
+                Date date = simpleDateFormat.parse(value.substring(0, 12));
+                return new Timestamp(date.getTime()).getTime();
+            }
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Unexpected time format : " + value + ".");
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:266]: Not need append `.` at the end of exception message.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java [hunk 6]
```diff
@@ -106,4 +245,26 @@ public static String formatDate(Date date) {
         return sdf.format(date);
     }
 
+    public static long parseDateTime(String value) {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:248]: Java doc, please.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java [hunk 7]
```diff
@@ -106,4 +245,26 @@ public static String formatDate(Date date) {
...
+        try {
+            if (value.length() < 8) {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:250]: What do the 8 and 9 mean here?
Suggested adding a comment for them.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java [hunk 8]
```diff
@@ -144,4 +169,60 @@ private List<InLongMessage> transformMessageObjs(ClientContext context, InLongTo
...
+                String epoch = attributes.get(INLONGMSG_ATTR_TIME_DT).trim();
+                msgTime = Long.parseLong(epoch);
+            } else {
+                throw new IllegalArgumentException(
+                        "Could not find " + INLONGMSG_ATTR_TIME_T
+                                + " or " + INLONGMSG_ATTR_TIME_DT + " in attributes!");
+            }
+
+            String srcIp = Optional.ofNullable(attributes.get(INLONGMSG_ATTR_NODE_IP))
+                    .orElse("127.0.0.1");
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:208]: Suggested using a default constant for `127.0.0.1`.

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java:179]: fixed, rename it to **_newAttrBuilder_**

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java (hunk 5 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:266]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java (hunk 7 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:250]: Add the reason in the java doc, thx.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java (hunk 6 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/util/StringUtil.java:248]: fixed, thx

File: inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/test/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializerTest.java:102]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java (hunk 8 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/decode/MessageDeserializer.java:208]: fixed, thx

**Review by ciscozhou — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**